src/condor_schedd.V6/dedicated_scheduler.cpp

```diff
@@ -3723,10 +3723,7 @@ DedicatedScheduler::setScheduler( ClassAd* job_ad )
 		return false;
 	}
 
-	while( SetAttributeString(cluster, proc, ATTR_SCHEDULER,
-							  ds_name, NONDURABLE) ==  0 ) {
-		proc++;
-	}
+	SetAttributeString(cluster, proc, ATTR_SCHEDULER,ds_name, NONDURABLE);
```

Comment:
This is a change in functionality. Is it correct?

Reply:
Ah, it looks like the only caller does the iterating over all procs already. Can we just inline the function and remove the extra lookups of ClusterId and ProcId?

Reply:
My thought was it might be more efficient to look up to see if we've already set `Scheduler` to this same value, and not reset it, but never got around to that.  If so, maybe it deserves its own function.

Reply:
If we keep the function, can we change the params to be cluster and proc, which the sole caller already has?

Reply:
Not required. Approved as-is.